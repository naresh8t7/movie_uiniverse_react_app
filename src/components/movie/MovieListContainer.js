import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import * as movieAction from '../../action/MovieAction';
import MovieList from './MovieList';
import MovieGraph from './MovieGraph'



export class MovieListContainer extends React.Component {

    constructor() {
        super();

        this.state = {selectedMovieTitle: undefined};

        this.handleAddMovie = this.handleAddMovie.bind(this);
        this.handleEditMovie = this.handleEditMovie.bind(this);
        this.handleDelete = this.handleDelete.bind(this);
        this.handleRowSelect = this.handleRowSelect.bind(this);
    }


    componentDidMount() {
        this.props.action.getMoviesAction()
            .catch(error => {
                toastr.error(error);
            });
    }



    handleAddMovie() {
        this.props.history.push('/movie');
    }



    handleEditMovie() {
        const selectedMovieTitle = this.state.selectedMovieTitle;

        if (selectedMovieTitle) {
            this.setState({selectedMovieTitle: undefined});
            this.props.history.push(`/movie/${selectedMovieTitle}`);
        }        
    }



    handleDelete() {
        const selectedMovieTitle = this.state.selectedMovieTitle;

        if (selectedMovieTitle) {
            this.setState({selectedMovieTitle: undefined});
            this.props.action.deleteMovieAction(selectedMovieTitle)
                .catch(error => {
                    toastr.error(error);
                });
        }
    }



    handleRowSelect(row, isSelected) {
        if (isSelected) {
            this.setState({selectedMovieTitle: row.id});
        }
    }



    render() {
        const { movies } = this.props;

        if (!movies) {
            return (
                <div>Loading...</div>
            );
        }

        return (
            <div className="container-fluid">
                <div className="row mt-3">
                    <div className="col">
                        <h1>Movie Universe</h1>
                    </div>
                </div>

                <div className="row mt-3">
                    <div className="col">
                        <div className="btn-group" role="group">
                            <button
                                type="button"
                                className="btn btn-primary"
                                onClick={this.handleAddMovie}
                            >
                                <i className="fa fa-plus" aria-hidden="true"/> New
                            </button>

                            <button
                                type="button"
                                className="btn btn-warning ml-2"
                                onClick={this.handleEditMovie}
                            >
                                <i className="fa fa-pencil" aria-hidden="true"/> Edit
                            </button>                                

                            <button
                                type="button"
                                className="btn btn-danger ml-2"
                                onClick={this.handleDelete}
                            >
                                <i className="fa fa-trash-o" aria-hidden="true" onClick={this.handleDelete}/> Delete
                            </button>
                        </div>
                    </div>
                </div>

                <div className="row">
                    <div className="col">
                        <MovieList movies={movies} handleRowSelect={this.handleRowSelect}/>
                    </div>
                </div>

                <div className="row">
                    <div className="col">
                        <MovieGraph movies={movies} />
                    </div>
                </div>
            </div>
        );
    }
}



const mapStateToProps = state => ({
    movies: state.moviesReducer.movies
});



const mapDispatchToProps = dispatch => ({
    action: bindActionCreators(movieAction, dispatch)

});



export default connect(mapStateToProps, mapDispatchToProps)(MovieListContainer);
