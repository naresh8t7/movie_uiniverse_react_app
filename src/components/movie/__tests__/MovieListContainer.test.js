import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import createMemoryHistory from 'history/createMemoryHistory';
import { MovieListContainer } from '../MovieListContainer';


describe('MovieListContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            movies: [
                { title: 'movie' }
            ],
            action: { getMoviesAction: jest.fn().mockReturnValue(Promise.resolve('')) },
            history: createMemoryHistory()
        };

        const wrapper = shallow(<MovieListContainer {...props} />);

        expect(wrapper).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();

        expect(wrapper.find('button')).toHaveLength(3);
    });

  
});

