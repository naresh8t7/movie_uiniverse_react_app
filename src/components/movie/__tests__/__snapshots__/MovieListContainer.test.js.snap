// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListContainer.test.js renders without crashing 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h1>
        Movie Universe
      </h1>
    </div>
  </div>
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-plus"
          />
           New
        </button>
        <button
          className="btn btn-warning ml-2"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-pencil"
          />
           Edit
        </button>
        <button
          className="btn btn-danger ml-2"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-trash-o"
            onClick={[Function]}
          />
           Delete
        </button>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <MovieList
        handleRowSelect={[Function]}
        movies={
          Array [
            Object {
              "title": "movie",
            },
          ]
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <MovieGraph
        movies={
          Array [
            Object {
              "title": "movie",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
